src/App.js

```diff
@@ -61,53 +61,46 @@ izaberiAutora(autor){
   }
 
   filtriraj =()=> {
-    const jezik = this.state.engleski ? 'en' : 'tekst'
+    const jezik = this.state.jezik
     const filtrirano = this.state.citati.filter(citat =>
       (citat.autor === this.state.autor || this.state.autor === '')
-      && (jezik in citat)
-      && citat[jezik].toLowerCase().includes(this.state.tekst.toLowerCase())
+      && citat[jezik]
+      && citat[jezik].toLowerCase().includes(this.state.fraza.toLowerCase())
     )
     this.setState(() => ({filtrirano}))
   }
 
-  changeToEng = () => {
+  changeLang = (lang) => {
     this.setState({
-      engleski: true
-    })
-  }
-  changeToSrb = () => {
-    this.setState({
-      engleski: false
+      jezik: lang
     })
   }
+
   render() {
-    const citati = this.state.filtrirano.map((citat, i) =>{
-      let tekst;
-      if(this.state.engleski && citat.en){
-        tekst = citat.en
-      }
-      if(!this.state.engleski && citat.tekst){
-        tekst = citat.tekst
-      }
-      return tekst ? <Quote className="not" key={i} tekst={tekst} autor={citat.autor} slika={this.state.slikeAutora.get(citat.autor)} /> : ''
+    const citati = this.state.filtrirano.map((citat, i) => {
+      const tekst = citat[this.state.jezik]
+      return tekst ? 
+        <Quote className="not" key={i} tekst={tekst} autor={citat.autor} /> 
```

Comment:
stavljanje index-a kao key nije dobra ideja. https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318